web/src/components/forms/blank-project-form.tsx

```diff
@@ -119,8 +120,8 @@ sample_table: samples.csv
           {...register('project_name', {
             required: true,
             pattern: {
-              value: /^\S+$/,
-              message: 'No spaces allowed.',
+              value: /^[a-zA-Z0-9_-]+$/,
+              message: "Project Name must contain only alphanumeric characters, '-', or '_'.",
```

Comment:
This is awesome, I didn't know this was possible with `react-hook-form`!

Reply:
```
// instantiate form
const {
    reset: resetForm,
    register,
    control,
    watch,
    setValue,
    formState: { isValid, errors },
  } = useForm<FromFileInputs>({
    mode: 'onChange',  <<<---------- had to add this for it to work
    defaultValues: {
      is_private: false,
      pep_schema: 'pep/2.1.0',
      namespace: defaultNamespace || user?.login || '',
    },
  });
```

So I added that mode: 'onChange' line to make it work but wasn't sure if it would cause lag if it started checking after every single input. Is this negligible with react/our use case or is it something we have to worry about?